src/app/portlets/content-types/main/content-types.component.ts

```diff
@@ -24,133 +26,107 @@ import { MessageService } from '../../../api/services/messages-service';
+                        this.createContentType($event);
                     },
-                    {
-                        command: $event => {
-                            this.createContentType('file', $event);
-                        },
-                        icon: 'fa-file-o',
-                        label: this.i18nMessages['contenttypes.content.file']
+                    model: []
+                }
+            };
+            val[1].filter(r => ['RECENT_CONTENT', 'RECENT_WIDGET'].indexOf(r.name) === -1).forEach(obj => {
+                this.actionHeaderOptions.primary.model.push({
+                    command: $event => {
+                        this.createContentType(obj.name.toLocaleLowerCase(), $event);
                     },
-                    {
-                        command: $event => {
-                            this.createContentType('page', $event);
-                        },
-                        icon: 'fa-file-text-o',
-                        label: this.i18nMessages['contenttypes.content.page']
-                    },
-                    {
-                        command: $event => {
-                            this.createContentType('persona', $event);
-                        },
-                        icon: 'fa-user',
-                        label: this.i18nMessages['contenttypes.content.persona']
-                    }
-                ]
-            }
-        };
-
-        this.contentTypeColumns = [
-            {
-                fieldName: 'name',
-                header: this.i18nMessages['contenttypes.fieldname.structure.name'],
-                icon: (item: any): string => this.contentTypesInfoService.getIcon(item.baseType),
-                sortable: true
-            },
-            {
-                fieldName: 'variable',
-                header: this.i18nMessages['contenttypes.content.variable'],
-                sortable: true
-            },
-            {
-                fieldName: 'description',
-                header: this.i18nMessages['contenttypes.form.label.description'],
-                sortable: true
-            },
-            {
-                fieldName: 'nEntries',
-                header: this.i18nMessages['contenttypes.fieldname.entries'],
-                width: '7%'
-            },
-            {
-                fieldName: 'modDate',
-                format: 'date',
-                header: 'Last Edit Date',
-                sortable: true,
-                width: '13%'
-            }
-        ];
-
-        this.rowActions = [
-            {
-                label: 'Remove',
-                icon: 'fa-trash',
-                command: (item) => this.removeConfirmation(item)
-            }
-        ];
+                    icon: this.contentTypesInfoService.getIcon(obj.name),
+                    label: this.i18nMessages['contenttypes.content.' + obj.name.toLocaleLowerCase()]
+                });
+            });
+            this.contentTypeColumns = [
```

Comment:
Move this to a external function so ```ngOnInit``` it's not that big and @freddyucv don't frak out, like 

```this.contentTypeColumns = this.getTypeColumns()```